extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/SingleValueScrutinizer.java

```diff
@@ -44,6 +49,24 @@
 public class SingleValueScrutinizer extends EditScrutinizer {
 
     public static final String type = "single-valued-property-added-more-than-once";
+    public static String SINGLE_VALUE_CONSTRAINT_QID = "Q19474404";
+    public static String SINGLE_VALUE_CONSTRAINT_EXCEPTION = "P2303";
+
+    class SingleValueConstraint {
```

Comment:
I think all exceptions can accept constraints, so how about we pull the parsing code in this class to a common superclass of all the inner classes you have been introducing?

Reply:
Sure, we can make a constraint class that will be extended by these inner classes. And check what all parameters can be added over there other than exceptions that are generally common to all property constraints.

Reply:
Great! Constraint status is another one.